### PR TITLE
Add parseInBatches Shapefile method

### PR DESCRIPTION
**Pull request by kylebarron**
This adds a `parseInBatches` method to the `SHPLoader`. For now, this is still non-streaming, but should be streaming in the future.

The output of each batch is:
```
{
  header: {
    length: 676,
    type: 5,
    bbox: {
      minX: 0,
      minY: 0,
      minZ: 0,
      minM: 0,
      maxX: 5,
      maxY: 5,
      maxZ: 0,
      maxM: 0
    }
  },
  features: [
    { positions: [Object], indices: [Object] },
    { positions: [Object], indices: [Object] },
    { positions: [Object], indices: [Object] }
  ]
}
```

Relies on #820.

### CODE REVIEW

File: modules/shapefile/src/shp-loader.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import {concatenateChunksAsync} from '@loaders.gl/core';
```
Comment (ibgreen) [inline modules/shapefile/src/shp-loader.js:1]: Can this be imported from `loader-utils`? Loader modules should not depend on core.

File: modules/shapefile/test/shp-loader.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  let batch;
+  let batchCount = 0;
+  let rowCount = 0;
+  for await (batch of iterator) {
+    batchCount++;
+    rowCount += batch.features.length;
+  }
+
+  t.equal(batchCount, 1, 'Correct number of batches received');
+  t.equal(rowCount, 3, 'Correct number of row received');
```
Comment (ibgreen) [inline modules/shapefile/test/shp-loader.spec.js:20]: rows

File: modules/shapefile/test/shp-loader.spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const SHAPEFILE_POLYGON_PATH = '@loaders.gl/shapefile/test/data/shapefile-js/polygons.shp';
+
+test('SHPLoader#loadInBatches polygons', async t => {
+  const iterator = await loadInBatches(SHAPEFILE_POLYGON_PATH, SHPLoader);
+  t.ok(isIterator(iterator) || isAsyncIterable(iterator), 'loadInBatches returned iterator');
+
+  let batch;
+  let batchCount = 0;
+  let rowCount = 0;
+  for await (batch of iterator) {
```
Comment (ibgreen) [inline modules/shapefile/test/shp-loader.spec.js:14]: `const batch of iterator`?

File: modules/shapefile/src/shp-loader.js [hunk 4]
```diff
@@ -24,5 +25,11 @@ export const SHPWorkerLoader = {
 export const SHPLoader = {
   ...SHPWorkerLoader,
   parse: async (arrayBuffer, options) => parseSHP(arrayBuffer),
-  parseSync: parseSHP
+  parseSync: parseSHP,
+  parseInBatches
 };
+
+async function* parseInBatches(asyncIterator, options) {
```
Comment (ibgreen) [inline modules/shapefile/src/shp-loader.js:32]: Maybe add TODO that this should be handled by `parseInBatches`. In fact you might want to try your hand at making that change as well - in a separate PR of course.

File: modules/shapefile/src/shp-loader.js (hunk 1 above)
Comment (kylebarron) [reply modules/shapefile/src/shp-loader.js:1]: `concatenateChunksAsync` doesn't currently exist in `loader-utils`.

File: modules/shapefile/test/shp-loader.spec.js (hunk 3 above)
Comment (kylebarron) [reply modules/shapefile/test/shp-loader.spec.js:14]: I copied this from the JSONLoader spec

File: modules/shapefile/src/shp-loader.js (hunk 1 above)
Comment (kylebarron) [reply modules/shapefile/src/shp-loader.js:1]: Should I copy it to `loader-utils`?

Comment (kylebarron) [reply modules/shapefile/src/shp-loader.js:1]: There's a copy in `loader-utils` which also currently returns a `Uint8Array`:
https://github.com/visgl/loaders.gl/blob/82872c693e9ef86801b11db9f5f81ea623601583/modules/loader-utils/src/lib/binary-utils/memory-copy-utils.js#L20-L30

**Pull request merged**